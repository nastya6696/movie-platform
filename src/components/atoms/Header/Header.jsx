import {LogoIcon} from "../LogoIcon";

import styles from './styles.module.scss'

export const Header = ({btnClickHandler, btnInfo }) => {
  const {name = '', type = 'addMovie'} = btnInfo;
  return (
    <div className={styles.header}>
      <LogoIcon boldPart="netflix" thinPart="roulette"/>
      <button onClick={btnClickHandler} className={styles[type]}>{name}</button>
    </div>
  )
}
