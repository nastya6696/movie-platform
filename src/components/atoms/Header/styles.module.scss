@import "../../../assets/scss/common";

.header {
  display: flex;
  justify-content: space-between;
  padding: 15px 60px;
}

.addMovie {
  width: 177px;
  height: 46px;
  border-radius: 4px;
  color: $red;
  background-color: $grey-background-30-transparent;
  border: none;
  @include fontSizes(20px, 25px);
  padding: 11px 18px;
}

.search {
  width: 30px;
  height: 30px;
  background-image: url("../../../assets/svg/searchBtn.svg");
  background-size: cover;
  background-repeat: no-repeat;
  border: none;
  background-color: inherit;
}
