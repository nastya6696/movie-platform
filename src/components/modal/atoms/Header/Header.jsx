import styles from './styles.module.scss';

export const Header = ({title}) => (
  <h2 className={styles.title}>{title}</h2>
)
