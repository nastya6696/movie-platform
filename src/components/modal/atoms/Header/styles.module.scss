@import "src/assets/scss/common";

.title {
  @include fontSizes(40px, 48px);
  font-weight: 300;
  color: $white;
}
