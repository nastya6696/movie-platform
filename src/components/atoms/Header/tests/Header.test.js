import { create } from 'react-test-renderer';
import {fireEvent, render, screen} from '@testing-library/react';
import { Header } from '../Header';

jest.mock('../../LogoIcon', () => ({
  LogoIcon: (({ boldPart, thinPart }) => `
     (--- LogoIcon ---): props {
     boldPart: '${boldPart}',
     thinPart: '${thinPart}'
    }`
  )
}));

describe('Header', () => {
  it('should create simple Header component', () => {
    const btnClickHandler = jest.fn();
    const testBtnInfo = {
      name: 'testName',
      type: 'testType'
    }

    const component = create(
      <Header btnClickHandler={btnClickHandler} btnInfo={testBtnInfo}/>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should call addBtn click handler', () => {
    const btnClickHandler = jest.fn();
    const testBtnInfo = {
      name: 'testName',
      type: 'testType'
    }

    render(
      <Header btnClickHandler={btnClickHandler} btnInfo={testBtnInfo} />
    );
    expect(btnClickHandler).not.toHaveBeenCalled();

    fireEvent.click(screen.getByText(testBtnInfo.name));

    expect(btnClickHandler).toHaveBeenCalledTimes(1);
  })
});
