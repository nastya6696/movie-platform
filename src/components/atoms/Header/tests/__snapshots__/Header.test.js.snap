// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should create simple Header component 1`] = `
<div
  className="header"
>
  
     (--- LogoIcon ---): props {
     boldPart: 'netflix',
     thinPart: 'roulette'
    }
  <button
    className="testType"
    onClick={[MockFunction]}
  >
    testName
  </button>
</div>
`;
